VISION.md

```diff
@@ -0,0 +1,98 @@
+# A vision for Gumption
+
+## Things I wanna do with `gum`
+
+Primarily, replace my use of other tools. So my flow now is:
+
+-   Look at all my "tracked" branches easily in some kind tree structure
+-   Create a new branch off the current branch, add all edited files and push to
+    new branch all in one command
+-   Quick command to commit all files to current branch
+-   Switch to a branch I didn't create, start tracking it
+-   Rebase current branch onto another target branch
+-   Alias command for `git rebase --continue`
+-   Alias command for `git add .` (while rebasing but also in general)
+-   Sync command. Takes all tracked branches, rebase them on main. Any branch
+    not branched off main, we rebase onto parent recursively. If a PR for that
+    branch is closed, delete the branch
+-   Submit a branch straight to github. If it's stacked on something then
+
+## High level behaviour
+
+Only some branches are "tracked". Trying to handle all branches is a bad idea
+imo and users should opt-in branches explicitly to be handled in this way.
+
+We will keep track of which branches are being handled this way in some system
+file.
+
+## Commands
+
+1. `hop` - list all branches, not just tracked ones, let's you search with
+   partial string and then menu select a branch to hop to, basically git switch
+   with better search & UI
+2. `continue` - basically just `git rebase --continue`
+3. `all` - basically just `git add .`
+4. `onto` - opens a menu to select a branch, rebases your current branch "onto"
+   that chosen branch
+5. `track` - start "tracking" current branch
+6. `go` / `send` (idk what to call this) - basically `git commit -am ""` and
```

Comment:
Not sure either, but if it's just committing files, then `go` or `send` feels off? Could be something around prepping for commit? `prep`, `ready`, `hold`, `add`?

Reply:
Made a new structure for the command names. Idea is to have them be memorable words and then just have abbreviated aliases